### ISSUE TITLE
Latest wasm test suite failures

### ISSUE THREAD

**Issue by nomeata** [dfinity/winter#4]
If I run `cabal test` with the latest test suite, I get the following log:

```
~/dfinity/winter $ WASM_SPEC_TESTS=../wasm-spec/test/core/ cabal new-test 
Build profile: -w ghc-8.6.5 -O1
In order, the following will be built (use -v for more details):
 - winter-1.0.0 (lib) (file src/Wasm/Exec/Eval.hs changed)
 - winter-1.0.0 (test:wasm-test-suite) (dependency rebuilt)
Preprocessing library for winter-1.0.0..
Building library for winter-1.0.0..
[26 of 27] Compiling Wasm.Exec.Eval   ( src/Wasm/Exec/Eval.hs, /home/jojo/dfinity/winter/dist-newstyle/build/x86_64-linux/ghc-8.6.5/winter-1.0.0/build/Wasm/Exec/Eval.o )
Preprocessing test suite 'wasm-test-suite' for winter-1.0.0..
Building test suite 'wasm-test-suite' for winter-1.0.0..
Linking /home/jojo/dfinity/winter/dist-newstyle/build/x86_64-linux/ghc-8.6.5/winter-1.0.0/t/wasm-test-suite/build/wasm-test-suite/wasm-test-suite ...
Running 1 test suites...
Test suite wasm-test-suite: RUNNING...
Using wasm spec test directory: ../wasm-spec/test/core/
main
  property
    compressible unsigned 32-bit integer:                OK
      +++ OK, passed 100 tests.
    compressible signed 32-bit integer:                  OK
      +++ OK, passed 100 tests.
    compressible unsigned 64-bit integer:                OK
      +++ OK, passed 100 tests.
    compressible signed 64-bit integer:                  OK
      +++ OK, passed 100 tests.
  spec
    ../wasm-spec/test/core//memory_redundancy.wast:      OK
    ../wasm-spec/test/core//type.wast:                   OK
    ../wasm-spec/test/core//if.wast:                     OK (0.01s)
    ../wasm-spec/test/core//imports.wast:                I32 13 : I32Type
I32 14 : I32Type
F32 42.0 : F32Type
I32 13 : I32Type
I32 13 : I32Type
F32 13.0 : F32Type
I32 13 : I32Type
F64 25.0 : F64Type
F64 53.0 : F64Type
F64 24.0 : F64Type
F64 24.0 : F64Type
F64 24.0 : F64Type
OK (0.05s)
    ../wasm-spec/test/core//start.wast:                  I32 1 : I32Type
I32 2 : I32Type
OK
    ../wasm-spec/test/core//utf8-import-module.wast:     OK (0.02s)
    ../wasm-spec/test/core//globals.wast:                OK (0.02s)
    ../wasm-spec/test/core//float_misc.wast:             OK (0.03s)
    ../wasm-spec/test/core//f32.wast:                    OK (0.09s)
    ../wasm-spec/test/core//block.wast:                  OK
    ../wasm-spec/test/core//int_exprs.wast:              OK (0.03s)
    ../wasm-spec/test/core//store.wast:                  OK
    ../wasm-spec/test/core//token.wast:                  OK
    ../wasm-spec/test/core//unreachable.wast:            OK
    ../wasm-spec/test/core//br_if.wast:                  OK
    ../wasm-spec/test/core//return.wast:                 OK
    ../wasm-spec/test/core//select.wast:                 OK
    ../wasm-spec/test/core//int_literals.wast:           OK
    ../wasm-spec/test/core//custom.wast:                 OK
    ../wasm-spec/test/core//unreached-invalid.wast:      OK
    ../wasm-spec/test/core//binary.wast:                 OK
    ../wasm-spec/test/core//break-drop.wast:             OK
    ../wasm-spec/test/core//align.wast:                  OK (0.04s)
    ../wasm-spec/test/core//memory_size.wast:            OK (0.04s)
    ../wasm-spec/test/core//elem.wast:                   OK (0.03s)
    ../wasm-spec/test/core//local_tee.wast:              FAIL
      test/Spec.hs:28:
      HUnitFailure (Just (SrcLoc {srcLocPackage = "main", srcLocModule = "Spec", srcLocFile = "test/Spec.hs", srcLocStartLine = 28, srcLocStartCol = 16, srcLocEndLine = 28, srcLocEndCol = 27})) "as-unary-operand [F32 0.0] == [F32 NaN]\nexpected: Right [F32 NaN]\n but got: Right [F32 NaN]"
    ../wasm-spec/test/core//call_indirect.wast:          OK (1.07s)
    ../wasm-spec/test/core//fac.wast:                    OK
    ../wasm-spec/test/core//nop.wast:                    OK (0.04s)
    ../wasm-spec/test/core//traps.wast:                  OK
    ../wasm-spec/test/core//local_set.wast:              OK
    ../wasm-spec/test/core//br_table.wast:               OK (0.03s)
    ../wasm-spec/test/core//utf8-custom-section-id.wast: OK
    ../wasm-spec/test/core//br.wast:                     OK
    ../wasm-spec/test/core//left-to-right.wast:          OK
    ../wasm-spec/test/core//comments.wast:               OK
    ../wasm-spec/test/core//const.wast:                  FAIL
      Exception: user error (./test24053-472.wat:1:26: error: invalid literal "nan:0x7f_ffff"
      (module (func (f32.const nan:0x7f_ffff) drop))
                               ^^^^^^^^^^^^^
      )
    ../wasm-spec/test/core//loop.wast:                   OK
    ../wasm-spec/test/core//endianness.wast:             OK
    ../wasm-spec/test/core//memory_grow.wast:            OK (4.53s)
    ../wasm-spec/test/core//float_memory.wast:           OK
    ../wasm-spec/test/core//f32_cmp.wast:                OK (0.08s)
    ../wasm-spec/test/core//func.wast:                   OK
    ../wasm-spec/test/core//memory_trap.wast:            OK
    ../wasm-spec/test/core//func_ptrs.wast:              I32 83 : I32Type
OK
    ../wasm-spec/test/core//binary-leb128.wast:          OK
    ../wasm-spec/test/core//switch.wast:                 OK
    ../wasm-spec/test/core//local_get.wast:              OK
    ../wasm-spec/test/core//forward.wast:                OK
    ../wasm-spec/test/core//f64.wast:                    OK (0.10s)
    ../wasm-spec/test/core//unwind.wast:                 OK
    ../wasm-spec/test/core//call.wast:                   OK (0.86s)
    ../wasm-spec/test/core//labels.wast:                 OK (0.01s)
    ../wasm-spec/test/core//memory.wast:                 OK
    ../wasm-spec/test/core//skip-stack-guard-page.wast:  OK (0.05s)
    ../wasm-spec/test/core//linking.wast:                FAIL (0.03s)
      test/Spec.hs:28:
      HUnitFailure (Just (SrcLoc {srcLocPackage = "main", srcLocModule = "Spec", srcLocFile = "test/Spec.hs", srcLocStartLine = 28, srcLocStartCol = 16, srcLocEndLine = 28, srcLocEndCol = 27})) "get memory[0] [] == [I32 104]\nexpected: Right [I32 104]\n but got: Right [I32 0]"
    ../wasm-spec/test/core//utf8-import-field.wast:      OK
    ../wasm-spec/test/core//data.wast:                   OK (0.03s)
    ../wasm-spec/test/core//load.wast:                   OK
    ../wasm-spec/test/core//stack.wast:                  OK
    ../wasm-spec/test/core//utf8-invalid-encoding.wast:  OK
    ../wasm-spec/test/core//exports.wast:                OK (0.06s)
    ../wasm-spec/test/core//f64_cmp.wast:                OK (0.09s)

3 out of 67 tests failed (7.74s)
Test suite wasm-test-suite: FAIL
```

Did not investigate yet.

**Comment by nomeata** [comment]
I guess the floating-point related one should just be disabled as per comment
```
              -- jww (2018-11-03): We need more accurate floating-point support.
```

**Comment by nomeata** [comment]
Fixed in #10
